--- gcc_4.3.6/gcc/toplev.h.orig	2007-09-24 02:18:27.000000000 +0700
+++ gcc_4.3.6/gcc/toplev.h	2015-09-19 13:54:47.200589653 +0600
@@ -172,18 +172,6 @@ extern int floor_log2                  (
 #  define CLZ_HWI __builtin_clz
 #  define CTZ_HWI __builtin_ctz
 # endif
-
-extern inline int
-floor_log2 (unsigned HOST_WIDE_INT x)
-{
-  return x ? HOST_BITS_PER_WIDE_INT - 1 - (int) CLZ_HWI (x) : -1;
-}
-
-extern inline int
-exact_log2 (unsigned HOST_WIDE_INT x)
-{
-  return x == (x & -x) && x ? (int) CTZ_HWI (x) : -1;
-}
 #endif /* GCC_VERSION >= 3004 */
 
 /* Functions used to get and set GCC's notion of in what directory
