diff -urN linux-2.4.32/drivers/usb/Makefile linux-2.4.32-new/drivers/usb/Makefile
--- linux-2.4.32/drivers/usb/Makefile   2004-02-18 15:36:31.000000000 +0200
+++ linux-2.4.32-new/drivers/usb/Makefile       2006-02-01 21:57:21.000000000 +0200
@@ -75,6 +75,11 @@
 subdir-$(CONFIG_USB_OHCI)      += host
 ifeq ($(CONFIG_USB_OHCI),y)
        obj-y += host/usb-ohci.o
 endif
+
+subdir-$(CONFIG_USB_AHCI)      += host
+ifeq ($(CONFIG_USB_AHCI),y)
+       obj-y += host/usb-shci.o
+endif

 subdir-$(CONFIG_USB_SL811HS_ALT)       += host
