--- socat-1.6.0.0/xioinitialize.c.orig	2007-03-07 03:24:10.000000000 +0600
+++ socat-1.6.0.0/xioinitialize.c	2010-02-26 20:10:27.000000000 +0600
@@ -35,40 +35,6 @@ int xioinitialize(void) {
    assert(SHUT_WR==1);
    assert(SHUT_RDWR==2);
 
-   /* some assertions about termios */
-#if WITH_TERMIOS
-#ifdef CRDLY
-   assert(3 << opt_crdly.arg3  == CRDLY);
-#endif
-#ifdef TABDLY
-   assert(3 << opt_tabdly.arg3 == TABDLY);
-#endif
-   assert(3 << opt_csize.arg3  == CSIZE);
-   {
-      union {
-	 struct termios termarg;
-	 tcflag_t flags[4];
-#if HAVE_TERMIOS_ISPEED
-	 speed_t speeds[sizeof(struct termios)/sizeof(speed_t)];
-#endif
-      } tdata;
-      tdata.termarg.c_iflag = 0x12345678;
-      tdata.termarg.c_oflag = 0x23456789;
-      tdata.termarg.c_cflag = 0x3456789a;
-      tdata.termarg.c_lflag = 0x456789ab;
-      assert(tdata.termarg.c_iflag == tdata.flags[0]);
-      assert(tdata.termarg.c_oflag == tdata.flags[1]);
-      assert(tdata.termarg.c_cflag == tdata.flags[2]);
-      assert(tdata.termarg.c_lflag == tdata.flags[3]);
-#if HAVE_TERMIOS_ISPEED
-      tdata.termarg.c_ispeed = 0x56789abc;
-      tdata.termarg.c_ospeed = 0x6789abcd;
-      assert(tdata.termarg.c_ispeed == tdata.speeds[ISPEED_OFFSET]);
-      assert(tdata.termarg.c_ospeed == tdata.speeds[OSPEED_OFFSET]);
-#endif
-   }
-#endif
-
    /* these dependencies required in applyopts() for OFUNC_FCNTL */
    assert(F_GETFD == F_SETFD-1);
    assert(F_GETFL == F_SETFL-1);
