--- ser2net-2.7/dataxfer.c.orig	2009-10-23 19:27:54.000000000 +0700
+++ ser2net-2.7/dataxfer.c	2010-03-11 19:11:12.000000000 +0600
@@ -472,6 +472,7 @@ handle_dev_fd_read(int fd, void *data)
 	return;
     } else if (port->dev_to_tcp_buf_count == 0) {
 	/* The port got closed somehow, shut it down. */
+	syslog(LOG_ERR, "EOF readed on port %s: %m", port->portname);
 	shutdown_port(port, "closed port");
 	return;
     }
