diff -ruN tapi/src/drv_tapi.h tapi_patched/src/drv_tapi.h
--- tapi/src/drv_tapi.h	2007-10-22 13:31:33.000000000 +0000
+++ tapi_patched/src/drv_tapi.h	2008-01-17 13:34:40.000000000 +0000
@@ -143,8 +143,6 @@
 /* Declarations for debug interface */
 DECLARE_TRACE_GROUP  (TAPI_DRV);
 DECLARE_LOG_GROUP    (TAPI_DRV);
-/* global high level driver context used in system interface only */
-extern struct _IFX_TAPI_HL_DRV_CTX gHLDrvCtx [TAPI_MAX_LL_DEVICES];
 /* Counter of registered low level device drivers */
 extern int ifx_tapi_drvctx_count;
 
@@ -1019,4 +1017,7 @@
                                    (IFX_int32_t Major);
 extern IFX_int32_t IFX_TAPI_PKT_RTP_PT_Defaults (TAPI_CHANNEL *pChannel);
 
+/* global high level driver context used in system interface only */
+extern struct _IFX_TAPI_HL_DRV_CTX gHLDrvCtx [TAPI_MAX_LL_DEVICES];
+
 #endif  /* DRV_TAPI_H */
diff -ruN tapi/src/Makefile.am tapi_patched/src/Makefile.am
--- tapi/src/Makefile.am	2007-10-01 14:03:06.000000000 +0000
+++ tapi_patched/src/Makefile.am	2008-01-17 13:34:40.000000000 +0000
@@ -160,7 +160,7 @@
 
 # Extra rule for linux-2.6 kernel object
 if KERNEL_2_6
-drv_tapi_OBJS = "$(subst .c,.o,$(notdir $(drv_tapi_SOURCES)))"
+drv_tapi_OBJS = "$(subst .c,.o,$(drv_tapi_SOURCES))"
 
 drv_tapi.ko: $(drv_tapi_SOURCES) $(EXTRA_DIST)
 	@echo -e "Making Linux 2.6.x kernel object"
