diff -ruiN tapi_pure/src/drv_tapi_fxo.c tapi/src/drv_tapi_fxo.c
--- tapi_pure/src/drv_tapi_fxo.c	2009-02-04 20:28:48.000000000 +0600
+++ tapi/src/drv_tapi_fxo.c	2009-02-04 20:30:14.000000000 +0600
@@ -37,7 +37,7 @@
 
 /** maximum number of DAA channels TAPI can handle,
     (size of the DAA to TAPI_CH lookuptable) */
-#define IFX_TAPI_FXO_MAX_DAA_CH             16
+#define IFX_TAPI_FXO_MAX_DAA_CH             32
 
 
 static IFX_TAPI_DRV_CTX_DAA_t *gpDaaCtx = IFX_NULL;
diff -ruiN tapi_pure/src/drv_tapi.h tapi/src/drv_tapi.h
--- tapi_pure/src/drv_tapi.h	2009-02-04 20:28:48.000000000 +0600
+++ tapi/src/drv_tapi.h	2009-02-04 20:28:48.000000000 +0600
@@ -143,8 +143,6 @@
 /* Declarations for debug interface */
 DECLARE_TRACE_GROUP  (TAPI_DRV);
 DECLARE_LOG_GROUP    (TAPI_DRV);
-/* global high level driver context used in system interface only */
-extern struct _IFX_TAPI_HL_DRV_CTX gHLDrvCtx [TAPI_MAX_LL_DEVICES];
 /* Counter of registered low level device drivers */
 extern int ifx_tapi_drvctx_count;
 
@@ -1019,4 +1017,7 @@
                                    (IFX_int32_t Major);
 extern IFX_int32_t IFX_TAPI_PKT_RTP_PT_Defaults (TAPI_CHANNEL *pChannel);
 
+/* global high level driver context used in system interface only */
+extern struct _IFX_TAPI_HL_DRV_CTX gHLDrvCtx [TAPI_MAX_LL_DEVICES];
+
 #endif  /* DRV_TAPI_H */
diff -ruiN tapi_pure/src/Makefile.am tapi/src/Makefile.am
--- tapi_pure/src/Makefile.am	2009-02-04 20:28:48.000000000 +0600
+++ tapi/src/Makefile.am	2009-02-04 20:28:48.000000000 +0600
@@ -160,7 +160,7 @@
 
 # Extra rule for linux-2.6 kernel object
 if KERNEL_2_6
-drv_tapi_OBJS = "$(subst .c,.o,$(notdir $(drv_tapi_SOURCES)))"
+drv_tapi_OBJS = "$(subst .c,.o,$(drv_tapi_SOURCES))"
 
 drv_tapi.ko: $(drv_tapi_SOURCES) $(EXTRA_DIST)
 	@echo -e "Making Linux 2.6.x kernel object"
