diff -ruiN vinetic_pure/2cpe/drv_vinetic_api.c vinetic/2cpe/drv_vinetic_api.c
--- vinetic_pure/2cpe/drv_vinetic_api.c	2008-04-29 16:35:42.000000000 +0700
+++ vinetic/2cpe/drv_vinetic_api.c	2008-04-29 16:35:42.000000000 +0700
@@ -213,6 +213,11 @@
    IFXOS_ASSERT(!(pCmd[0] & CMD1_BC));
    IFXOS_ASSERT(nCount <= (MAX_PACKET_WORD - CMD_HEADER_CNT));
 
+	/* tag__ BUG_3 - correct assertion value */
+	if(pCmd[0] & 0x2000){
+		pCmd[0] &= ~(0x2000);
+	}
+
    /* if read command issued just write command words */
    if ((pCmd[0] & CMD1_RD) == 0)
    {
@@ -347,6 +352,11 @@
 
    IFXOS_ASSERT(!(pCmd[0] & CMD1_BC));
 
+	/* tag__ BUG_3 - correct assertion value */
+	if(pCmd[0] & 0x2000){
+		pCmd[0] &= ~(0x2000);
+	}
+
    /* protect against concurrent task access and VINETIC interrupts */
    VIN_HOST_PROTECT(pDev);
 
diff -ruiN vinetic_pure/2cpe/drv_vinetic_gpio.c vinetic/2cpe/drv_vinetic_gpio.c
--- vinetic_pure/2cpe/drv_vinetic_gpio.c	2008-04-29 16:35:42.000000000 +0700
+++ vinetic/2cpe/drv_vinetic_gpio.c	2008-04-29 16:35:42.000000000 +0700
@@ -801,4 +801,9 @@
 
    return err;
 }
+
+EXPORT_SYMBOL( VINETIC_GpioSet );
+EXPORT_SYMBOL( VINETIC_GpioGet );
+EXPORT_SYMBOL( VINETIC_GpioConfig );
+EXPORT_SYMBOL( VINETIC_GpioReserve );
 /* @} */
diff -ruiN vinetic_pure/2cpe/drv_vinetic_host.c vinetic/2cpe/drv_vinetic_host.c
--- vinetic_pure/2cpe/drv_vinetic_host.c	2008-04-29 16:35:42.000000000 +0700
+++ vinetic/2cpe/drv_vinetic_host.c	2008-04-29 16:35:42.000000000 +0700
@@ -927,7 +927,8 @@
       V2CPE_ERR_IEN_HW_ERR    |     /* EDSP hardware error */
       V2CPE_ERR_IEN_WD_FAIL   |     /* EDSP watchdog timeout */
       V2CPE_ERR_IEN_MCLK_FAIL |     /* MCLK failure */
-      V2CPE_ERR_IEN_SYNC_FAIL |     /* PLL clock out of sync */
+      /*  Remarked to prevent wrong SYNC_FAIL interrupt (BUG_8)  */
+      /* V2CPE_ERR_IEN_SYNC_FAIL |     PLL clock out of sync */
       V2CPE_ERR_IEN_EDSP_MIPS_OL |  /* EDSP MIPS overload */
       V2CPE_ERR_IEN_PCMB_CRASH|     /* PCM transmit highway B crash */
       V2CPE_ERR_IEN_PCMA_CRASH|     /* PCM transmit highway A crash */
diff -ruiN vinetic_pure/2cpe/Makefile.am vinetic/2cpe/Makefile.am
--- vinetic_pure/2cpe/Makefile.am	2008-04-29 16:35:42.000000000 +0700
+++ vinetic/2cpe/Makefile.am	2008-04-29 16:35:42.000000000 +0700
@@ -218,7 +218,7 @@
 
 # Extra rule for linux-2.6 kernel object
 if KERNEL_2_6
-drv_vinetic_OBJS = "$(subst .c,.o, $(notdir $(drv_vinetic_SOURCES)))"
+drv_vinetic_OBJS = "$(subst .c,.o, $(drv_vinetic_SOURCES))"
 
 drv_vinetic.ko:
 	@echo -e "Making Linux 2.6.x kernel object"
diff -ruiN vinetic_pure/src/drv_vinetic_linux.c vinetic/src/drv_vinetic_linux.c
--- vinetic_pure/src/drv_vinetic_linux.c	2008-04-29 16:35:42.000000000 +0700
+++ vinetic/src/drv_vinetic_linux.c	2008-04-29 16:35:42.000000000 +0700
@@ -1551,6 +1551,7 @@
 module_init(vinetic_module_init);
 module_exit(vinetic_module_exit);
 
+EXPORT_SYMBOL( VINETIC_OpenKernel );
 
 #endif /* LINUX */
 
