--- a/pci.ids	2010-01-19 20:18:08.000000000 +1100
+++ b/pci.ids	2010-03-14 12:15:36.000000000 +1100
@@ -1,8 +1,8 @@
 #
 #	List of PCI ID's
 #
-#	Version: 2010.01.19
-#	Date:    2010-01-19 10:18:08
+#	Version: 2010.03.01
+#	Date:    2010-03-01 03:15:01
 #
 #	Maintained by Martin Mares <mj@ucw.cz> and other volunteers from the
 #	PCI ID Project at http://pciids.sf.net/.
@@ -111,7 +111,7 @@
 07ca  AVerMedia Technologies Inc.
 	a301  AVerTV 301
 	b808  AVerTV DVB-T Volar (USB 2.0)
-07d0  ITT Space Systems Division
+07d0  ITT Geospatial Systems
 07e2  ELMEG Communication Systems GmbH
 0842  NPG, Personal Grand Technology
 # Nee Gemplus International, SA
@@ -1364,6 +1364,9 @@
 	5f57  R423 [Radeon X800XT (PCIE)]
 	6899  Cypress [Radeon HD 5800 Series]
 	68b8  Juniper [Radeon HD 5700 Series]
+	68c1  Redwood [Radeon HD 5600 Series]
+		1025 0347  Aspire 7740G
+	68e0  Manhattan [Mobility Radeon HD 5000 Series]
 	700f  PCI Bridge [IGP 320M]
 	7010  PCI Bridge [IGP 340M]
 	7100  R520 [Radeon X1800]
@@ -1643,6 +1646,9 @@
 		174b aa38  R700 Audio Device [Radeon HD 4000 Series]
 	aa50  Cypress HDMI Audio [Radeon HD 5800 Series]
 	aa58  Juniper HDMI Audio [Radeon HD 5700 Series]
+	aa60  Redwood HDMI Audio [Radeon HD 5600 Series]
+		1025 0347  Aspire 7740G
+	aa68  Manhattan HDMI Audio [Mobility Radeon HD 5000 Series]
 	ac00  Theater 600 Pro
 	cab0  AGP Bridge [IGP 320M]
 	cab2  RS200/RS200M AGP Bridge [IGP 340M]
@@ -3443,6 +3449,7 @@
 	8039  CXD3222 i.LINK Controller
 	8056  Rockwell HCF 56K modem
 	808a  Memory Stick Controller
+	81ce  SxS Pro memory card
 104e  Oak Technology, Inc
 	0017  OTI-64017
 	0107  OTI-107 [Spitfire]
@@ -3485,6 +3492,8 @@
 	3017  Unassigned Hitachi Shared FC Device 3017
 	301d  PCIe-to-PCIe Bridge with Virtualization IO Assist Feature
 	3020  FIVE-EX based Fibre Channel to PCIe HBA
+	302c  M001 PCI Express Switch Upstream Port
+	302d  M001 PCI Express Switch Downstream Port
 	3505  SH7751 PCI Controller (PCIC)
 	350e  SH7751R PCI Controller (PCIC)
 1055  Efar Microsystems
@@ -6201,7 +6210,7 @@
 	075c  MCP78S [GeForce 8200] LPC Bridge
 		1462 7508  K9N2GM-FIH
 		1849 075c  K10N78FullHD-hSLI R3.0 LPC Bridge
-	075d  nForce 750a LPC Bridge
+	075d  MCP78S [GeForce 8200] LPC Bridge
 		1043 82e8  M3N72-D
 	0760  MCP77 Ethernet
 		1043 82e8  M3N72-D
@@ -6320,17 +6329,26 @@
 	0871  C79 [GeForce 9200]
 	0872  C79 [GeForce G102M]
 	0873  C79 [GeForce G102M]
+	0876  ION VGA [GeForce 9400M]
 	087a  C79 [Quadro FX 470]
 	087d  ION VGA
 	087e  ION LE VGA
 	087f  C79 [Quadro FX 470M]
-	0a20  GT200 [GeForce GT 220]
-	0a28  GeForce GT 230M
-	0a60  GT200 [GeForce G210]
+	0a20  GT216 [GeForce GT 220]
+	0a23  GT218 [GeForce 210]
+	0a28  GT216 [GeForce GT 230M]
+	0a2a  GT216 [GeForce GT 230M]
+	0a34  GT216 [GeForce GT 240M]
+	0a60  GT218 [GeForce G210]
+	0a62  GT218 [GeForce 205]
+	0a63  GT218 [GeForce 310]
 	0a64  GT218 [ION]
-	0a65  GT200 [GeForce 210]
-# formerly known as NVIDIA Quadro NVS 170M
-	0a6c  NVIDIA NVS 3100M
+	0a65  GT218 [GeForce 210]
+	0a66  GT218 [GeForce 310]
+	0a6c  NVS 3100M
+	0a74  GT218 [GeForce G210M]
+	0a75  GT218 [GeForce 310M]
+	0a78  GT218GL [Quadro FX 380 LP]
 	0a80  MCP79 Host Bridge
 	0a81  MCP79 Host Bridge
 	0a82  MCP79 Host Bridge
@@ -6391,6 +6409,9 @@
 	0be2  High Definition Audio Controller
 	0be3  High Definition Audio Controller
 	0be4  High Definition Audio Controller
+	0ca3  GT215 [GeForce GT 240]
+	0ca8  GT215 [GeForce GTS 260M]
+	0ca9  GT215 [GeForce GTS 250M]
 10df  Emulex Corporation
 	1ae5  LP6000 Fibre Channel Host Adapter
 	e100  Proteus-X: LightPulse IOV Fibre Channel Host Adapter
@@ -6974,7 +6995,7 @@
 		10bd 0000  VT86C100A Fast Ethernet Adapter
 		1106 0100  VT86C100A Fast Ethernet Adapter
 		1186 1400  DFE-530TX rev A
-	3044  VT6306 Fire II IEEE 1394 OHCI Link Layer Controller
+	3044  VT6306/7/8 [Fire II(M)] IEEE 1394 OHCI Controller
 		0010 0001  IEEE 1394 4port DCST 1394-3+1B
 		1025 005a  TravelMate 290
 		1043 808a  A8V/A8N/P4P800 series motherboard
@@ -11206,6 +11227,7 @@
 	0811  Sentry5 External Interface Core
 	0816  BCM3302 Sentry5 MIPS32 CPU
 	1600  NetXtreme BCM5752 Gigabit Ethernet PCI Express
+		1028 01c1  Precision 490
 		1028 01c2  Latitude D620
 		103c 3015  PCIe LAN on Motherboard
 		107b 5048  E4500 Onboard
@@ -11408,6 +11430,7 @@
 	1690  NetXtreme BCM57760 Gigabit Ethernet PCIe
 	1691  NetLink BCM57788 Gigabit Ethernet PCIe
 	1692  NetLink BCM57780 Gigabit Ethernet PCIe
+		1025 033d  Aspire 7740G
 	1693  NetLink BCM5787M Gigabit Ethernet PCI Express
 		1025 0121  Aspire 5920G
 		103c 30c0  6710b
@@ -12448,11 +12471,22 @@
 	5449  SmartPCI561 Modem
 1641  MKNet Corp.
 1657  Brocade Communications Systems, Inc.
-	0013  425 4Gb/825 8Gb PCIe Dual port FC HBA
-	0014  BR-1020 FC/FCOE 10GB Eth / FC over Eth
-	0017  415 4Gb/815 8Gb PCIe FC HBA
+	0013  425/825/42B/82B 4Gbps/8Gbps PCIe dual port FC HBA
+		103c 1742  HP 82B 8Gbps dual port FC HBA
+		103c 1744  HP 42B 4Gbps dual port FC HBA
+		1657 0014  425/825 4Gbps/8Gbps PCIe dual port FC HBA
+	0014  1010/1020/1007 10Gbps CNA
+		1657 0014  1010/1020/1007 10Gbps CNA - FCOE
+		1657 0015  1010/1020/1007 10Gbps CNA - LL
+	0017  415/815/41B/81B 4Gbps/8Gbps PCIe single port FC HBA
+		103c 1741  HP 41B 4Gbps single port FC HBA
+		103c 1743  HP 81B 8Gbps single port FC HBA
+		1657 0014  415/815 4Gbps/8Gbps single port PCIe FC HBA
+	0021  804/807 8Gbps FC HBA
+		103c 1779  804 8Gbps FC HBA for HP Bladsystem c-class
+		1657 0014  8Gbps FC HBA
 # Same Device_ID used for 410 (1port) and 420 (2 port) HBAs.
-	0646  Brocade 400 4Gb PCIe FC HBA
+	0646  400 4Gbps PCIe FC HBA
 165a  Epix Inc
 	c100  PIXCI(R) CL1 Camera Link Video Capture Board [custom QL5232]
 	d200  PIXCI(R) D2X Digital Video Capture Board [custom QL5232]
@@ -13152,6 +13186,7 @@
 		1924 6202  SFN5122F-R2
 		1924 6203  SFN5122F-R3
 		1924 6204  SFN5122F-R4
+		1924 6205  SFN5122F-R5
 	0813  SFL9021 [Solarstorm]
 		1924 6100  SFN5121T-R0
 	6703  SFC4000 rev A iSCSI/Onload [Solarstorm]
@@ -13259,15 +13294,20 @@
 	a00d  PWRficient PCI-Express Internal Endpoint
 1966  Orad Hi-Tec Systems
 	1975  DVG64 family
-1969  Attansic Technology Corp.
+# nee Attansic Technology Corp.
+1969  Atheros Communications
 	1026  Atheros AR8121/AR8113/AR8114 PCI-E Ethernet Controller
 		1043 8304  P5KPL-CM Motherboard
 	1048  L1 Gigabit Ethernet Adapter
 		1043 8226  P5KPL-VM Motherboard
 	1062  Atheros AR8132 / L1c Gigabit Ethernet Adapter
+	1063  AR8131 Gigabit Ethernet
 	1066  L2c Gigabit Ethernet Adapter
 	1067  L1c Gigabit Ethernet Adapter
+	1073  AR8151 v1.0 Gigabit Ethernet
 	2048  L2 100 Mbit Ethernet Adapter
+	2060  AR8152 v1.1 Fast Ethernet
+	2062  AR8152 v2.0 Fast Ethernet
 196a  Sensory Networks Inc.
 	0101  NodalCore C-1000 Content Classification Accelerator
 	0102  NodalCore C-2000 Content Classification Accelerator
@@ -13300,6 +13340,10 @@
 	2387  SD/MMC Host Controller
 	2388  MS Host Controller
 	2389  xD Host Controller
+	2391  Standard SD Host Controller
+	2392  SD/MMC Host Controller
+	2393  MS Host Controller
+	2394  xD Host Controller
 1982  Distant Early Warning Communications Inc
 	1600  OX16C954 HOST-A
 	16ff  OX16C954 HOST-B
@@ -13413,8 +13457,10 @@
 	0e44  microEnable IV-GigE x4
 1aec  Wolfson Microelectronics
 1aed  Fusion-io
-	1003  ioDrive (NAND Flash Storage)
-	1005  ioDIMM3 320GB
+	1003  ioDimm3 (v1.2)
+	1005  ioDimm3
+	1006  ioXtreme
+	1007  ioXtreme Pro
 1aee  Caustic Graphics Inc.
 1af4  Qumranet, Inc.
 	1000  Virtio network device
@@ -13436,6 +13482,13 @@
 	2a2c  Dual DVB-S2-CI card
 	e2e4  Dual DVB-T/C-CI RF card
 1bad  ReFLEX CES
+1bb3  Bluecherry
+	4304  BC-04120A MPEG4 4 port video encoder / decoder
+	4309  BC-08240A MPEG4 4 port video encoder / decoder
+	4310  BC-16480A MPEG4 16 port video encoder / decoder
+	4e04  BC-04120A 4 port MPEG4 video encoder / decoder
+	4e09  BC-08240A 8 port MPEG4 video encoder / decoder
+	4e10  BC-16480A 16 port MPEG4 video encoder / decoder
 1c1c  Symphony
 	0001  82C101
 1d44  DPT
@@ -13999,6 +14052,7 @@
 	0042  Core Processor Integrated Graphics Controller
 	0043  Core Processor Secondary PCI Express Root Port
 	0044  Core Processor DRAM Controller
+		1025 0347  Aspire 7740G
 	0045  Core Processor PCI Express x16 Root Port
 	0046  Core Processor Integrated Graphics Controller
 	0047  Core Processor Secondary PCI Express Root Port
@@ -14029,6 +14083,7 @@
 	0089  WiMAX/WiFi Link 6050 Series
 		8086 1311  WiFi Link 6050 Series 2x2 6050AGN
 		8086 1316  WiFi Link 6050 Series 2x2 6050ABG
+	0101  Sandy Bridge PCI Express Root Port
 	0102  Sandy Bridge Integrated Graphics Controller
 	0103  Sandy Bridge Thermal Management Controller
 	0105  Sandy Bridge PCI Express Root Port
@@ -14098,7 +14153,28 @@
 		8086 01f7  SCRU32
 # uninitialized SRCU32 RAID Controller
 	061f  80303 I/O Processor
-	0800  SPI Ctrl 0
+	0800  Moorestown SPI Ctrl 0
+	0801  Moorestown SPI Ctrl 1
+	0802  Moorestown I2C 0
+	0803  Moorestown I2C 1
+	0804  Moorestown I2C 2
+	0805  Moorestown Keyboard Ctrl
+	0806  Moorestown USB Ctrl
+	0807  Moorestown SD Host Ctrl 0
+	0808  Moorestown SD Host Ctrl 1
+	0809  Moorestown NAND Ctrl
+	080a  Moorestown Audio Ctrl
+	080b  Moorestown ISP
+	080c  Moorestown Security Controller
+	080d  Moorestown External Displays
+	080e  Moorestown SCU IPC
+	080f  Moorestown GPIO Controller
+	0810  Moorestown Power Management Unit
+	0811  Moorestown OTG Ctrl
+	0812  Moorestown SPI Ctrl 2
+	0813  Moorestown SC DMA
+	0814  Moorestown LPE DMA
+	0815  Moorestown SSP0
 	0960  80960RP (i960RP) Microprocessor/Bridge
 	0962  80960RM (i960RM) Bridge
 	0964  80960RP (i960RP) Microprocessor/Bridge
@@ -14510,6 +14586,7 @@
 	10ec  82598EB 10-Gigabit AT CX4 Network Connection
 		8086 a01f  10-Gigabit CX4 Dual Port Server Adapter
 		8086 a11f  10-Gigabit CX4 Dual Port Server Adapter
+	10ed  82559 Ethernet Controller Virtual Function
 	10ef  82578DM Gigabit Network Connection
 	10f0  82578DC Gigabit Network Connection
 	10f1  82598EB 10-Gigabit AF Dual Port Network Connection
@@ -15932,10 +16009,10 @@
 		103c 30a3  Compaq nw8440
 		10f7 8338  Panasonic CF-Y5 laptop
 		17aa 2009  ThinkPad T60/R60 series
-	27bc  Tigerpoint LPC Controller
+	27bc  NM10 Family LPC Controller
 	27bd  82801GHM (ICH7-M DH) LPC Interface Bridge
 		1025 006c  9814 WKMI
-	27c0  82801GB/GR/GH (ICH7 Family) SATA IDE Controller
+	27c0  N10/ICH7 Family SATA IDE Controller
 		1028 01ad  OptiPlex GX620
 		1028 01df  PowerEdge SC440
 		1028 01e6  PowerEdge 860
@@ -15945,7 +16022,7 @@
 		1462 7236  945P Neo3-F Rev. 2.2 motherboard
 		1775 11cc  CC11/CL11
 		8086 544e  DeskTop Board D945GTP
-	27c1  82801GR/GH (ICH7 Family) SATA AHCI Controller
+	27c1  N10/ICH7 Family SATA AHCI Controller
 		1028 01df  PowerEdge SC440
 		1775 11cc  CC11/CL11
 		8086 5842  DeskTop Board D975XBX
@@ -15961,7 +16038,7 @@
 		103c 30a3  Compaq nw8440
 		17aa 200d  ThinkPad T60/R60 series
 	27c6  82801GHM (ICH7-M DH) SATA RAID Controller
-	27c8  82801G (ICH7 Family) USB UHCI Controller #1
+	27c8  N10/ICH7 Family USB UHCI Controller #1
 		1025 006c  9814 WKMI
 		1028 01ad  OptiPlex GX620
 		1028 01d7  XPS M1210
@@ -15976,7 +16053,7 @@
 		1775 11cc  CC11/CL11
 		17aa 200a  ThinkPad T60/R60 series
 		8086 544e  DeskTop Board D945GTP
-	27c9  82801G (ICH7 Family) USB UHCI Controller #2
+	27c9  N10/ICH 7 Family USB UHCI Controller #2
 		1025 006c  9814 WKMI
 		1028 01ad  OptiPlex GX620
 		1028 01d7  XPS M1210
@@ -15991,7 +16068,7 @@
 		1775 11cc  CC11/CL11
 		17aa 200a  ThinkPad T60/R60 series
 		8086 544e  DeskTop Board D945GTP
-	27ca  82801G (ICH7 Family) USB UHCI Controller #3
+	27ca  N10/ICH 7 Family USB UHCI Controller #3
 		1025 006c  9814 WKMI
 		1028 01ad  OptiPlex GX620
 		1028 01d7  XPS M1210
@@ -16006,7 +16083,7 @@
 		1775 11cc  CC11/CL11
 		17aa 200a  ThinkPad T60/R60 series
 		8086 544e  DeskTop Board D945GTP
-	27cb  82801G (ICH7 Family) USB UHCI Controller #4
+	27cb  N10/ICH 7 Family USB UHCI Controller #4
 		1025 006c  9814 WKMI
 		1028 01ad  OptiPlex GX620
 		1028 01d7  XPS M1210
@@ -16020,7 +16097,7 @@
 		1775 11cc  CC11/CL11
 		17aa 200a  ThinkPad T60/R60 series
 		8086 544e  DeskTop Board D945GTP
-	27cc  82801G (ICH7 Family) USB2 EHCI Controller
+	27cc  N10/ICH 7 Family USB2 EHCI Controller
 		1025 006c  9814 WKMI
 		1028 01ad  OptiPlex GX620
 		1028 01d7  XPS M1210
@@ -16034,20 +16111,20 @@
 		1775 11cc  CC11/CL11
 		17aa 200b  ThinkPad T60/R60 series
 		8086 544e  DeskTop Board D945GTP
-	27d0  82801G (ICH7 Family) PCI Express Port 1
+	27d0  N10/ICH 7 Family PCI Express Port 1
 		103c 309f  Compaq nx9420 Notebook
 		103c 30a3  Compaq nw8440
 		1775 11cc  CC11/CL11
-	27d2  82801G (ICH7 Family) PCI Express Port 2
+	27d2  N10/ICH 7 Family PCI Express Port 2
 		103c 309f  Compaq nx9420 Notebook
 		103c 30a3  Compaq nw8440
 		1775 11cc  CC11/CL11
-	27d4  82801G (ICH7 Family) PCI Express Port 3
+	27d4  N10/ICH 7 Family PCI Express Port 3
 		1775 11cc  CC11/CL11
-	27d6  82801G (ICH7 Family) PCI Express Port 4
+	27d6  N10/ICH 7 Family PCI Express Port 4
 		103c 30a3  Compaq nw8440
 		1775 11cc  CC11/CL11
-	27d8  82801G (ICH7 Family) High Definition Audio Controller
+	27d8  N10/ICH 7 Family High Definition Audio Controller
 		1025 006c  9814 WKMI
 		1028 01d7  XPS M1210
 		103c 309f  Compaq nx9420 Notebook
@@ -16069,7 +16146,7 @@
 		17aa 3802  Lenovo 3000 C200 audio [Realtek ALC861VD]
 		8086 1112  DeskTop Board D945GTP
 		8086 27d8  DeskTop Board D945GTP
-	27da  82801G (ICH7 Family) SMBus Controller
+	27da  N10/ICH 7 Family SMBus Controller
 		1025 006c  9814 WKMI
 		1028 01ad  OptiPlex GX620
 		1028 01d7  XPS M1210
@@ -16082,7 +16159,7 @@
 		17aa 200f  ThinkPad T60/R60 series
 		8086 544e  DeskTop Board D945GTP
 		8086 5842  DeskTop Board D975XBX
-	27dc  82801G (ICH7 Family) LAN Controller
+	27dc  N10/ICH 7 Family LAN Controller
 		8086 308d  DeskTop Board D945GTP
 	27dd  82801G (ICH7 Family) AC'97 Modem Controller
 	27de  82801G (ICH7 Family) AC'97 Audio Controller
@@ -16547,8 +16624,10 @@
 	2b1c  Nehalem-EX Memory Controller 1a
 	2b1e  Nehalem-EX Memory Controller 1b
 	2b20  Nehalem-EX Last Level Cache Coherence Engine 0
+	2b22  Nehalem-EX System Configuration Controller 3
 	2b24  Nehalem-EX Last Level Cache Coherence Engine 1
 	2b28  Nehalem-EX Last Level Cache Coherence Engine 2
+	2b2a  Nehalem-EX System Configuration Controller 4
 	2b2c  Nehalem-EX Last Level Cache Coherence Engine 3
 	2b30  Nehalem-EX Last Level Cache Coherence Engine 4
 	2b34  Nehalem-EX Last Level Cache Coherence Engine 5
@@ -16722,6 +16801,7 @@
 	2e45  4 Series Chipset HECI Controller
 	2e46  4 Series Chipset PT IDER Controller
 	2e47  4 Series Chipset Serial KT Controller
+	2e90  4 Series Chipset DRAM Controller
 	2e91  4 Series Chipset PCI Express Root Port
 	2e92  4 Series Chipset Integrated Graphics Controller
 	2e93  4 Series Chipset Integrated Graphics Controller
@@ -17074,6 +17154,7 @@
 	3b07  Mobile 5 Series Chipset LPC Interface Controller
 	3b08  5 Series Chipset LPC Interface Controller
 	3b09  Mobile 5 Series Chipset LPC Interface Controller
+		1025 0347  Aspire 7740G
 	3b0a  5 Series Chipset LPC Interface Controller
 	3b0b  Mobile 5 Series Chipset LPC Interface Controller
 	3b0c  5 Series Chipset LPC Interface Controller
@@ -17104,13 +17185,17 @@
 	3b26  5 Series/3400 Series Chipset 2 port SATA IDE Controller
 	3b28  5 Series/3400 Series Chipset 4 port SATA IDE Controller
 	3b29  5 Series/3400 Series Chipset 4 port SATA AHCI Controller
+		1025 0347  Aspire 7740G
 	3b2c  5 Series/3400 Series Chipset SATA RAID Controller
 	3b2d  5 Series/3400 Series Chipset 2 port SATA IDE Controller
 	3b2e  5 Series/3400 Series Chipset 4 port SATA IDE Controller
 	3b2f  5 Series/3400 Series Chipset 6 port SATA AHCI Controller
 	3b30  5 Series/3400 Series Chipset SMBus Controller
+		1025 0347  Aspire 7740G
 	3b32  5 Series/3400 Series Chipset Thermal Subsystem
+		1025 0347  Aspire 7740G
 	3b34  5 Series/3400 Series Chipset USB2 Enhanced Host Controller
+		1025 0347  Aspire 7740G
 	3b36  5 Series/3400 Series Chipset USB Universal Host Controller
 	3b37  5 Series/3400 Series Chipset USB Universal Host Controller
 	3b38  5 Series/3400 Series Chipset USB Universal Host Controller
@@ -17118,6 +17203,7 @@
 	3b3a  5 Series/3400 Series Chipset USB Universal Host Controller
 	3b3b  5 Series/3400 Series Chipset USB Universal Host Controller
 	3b3c  5 Series/3400 Series Chipset USB2 Enhanced Host Controller
+		1025 0347  Aspire 7740G
 	3b3e  5 Series/3400 Series Chipset USB Universal Host Controller
 	3b3f  5 Series/3400 Series Chipset USB Universal Host Controller
 	3b40  5 Series/3400 Series Chipset USB Universal Host Controller
@@ -17132,7 +17218,9 @@
 	3b50  5 Series/3400 Series Chipset PCI Express Root Port 8
 	3b53  5 Series/3400 Series Chipset VECI Controller
 	3b56  5 Series/3400 Series Chipset High Definition Audio
+		1025 0347  Aspire 7740G
 	3b64  5 Series/3400 Series Chipset HECI Controller
+		1025 0347  Aspire 7740G
 	3b65  5 Series/3400 Series Chipset HECI Controller
 	3b66  5 Series/3400 Series Chipset PT IDER Controller
 	3b67  5 Series/3400 Series Chipset KT Controller
@@ -17156,6 +17244,7 @@
 	4032  5400 Chipset IOxAPIC
 	4035  5400 Chipset FBD Registers
 	4036  5400 Chipset FBD Registers
+	4100  Moorestown Graphics and Video
 	4108  Tunnel Creek Integrated Graphics Controller
 	4109  Tunnel Creek Integrated Graphics Controller
 	410a  Tunnel Creek Integrated Graphics Controller
@@ -17482,14 +17571,14 @@
 	9622  Integrated RAID
 	9641  Integrated RAID
 	96a1  Integrated RAID
-	a000  Pineview DMI Bridge
-	a001  Pineview Integrated Graphics Controller
-	a002  Pineview Integrated Graphics Controller
-	a003  Pineview CHAPS counter
-	a010  Pineview DMI Bridge
-	a011  Pineview Integrated Graphics Controller
-	a012  Pineview Integrated Graphics Controller
-	a013  Pineview CHAPS counter
+	a000  N10 Family DMI Bridge
+	a001  N10 Family Integrated Graphics Controller
+	a002  N10 Family Integrated Graphics Controller
+	a003  N10 Family CHAPS counter
+	a010  N10 Family DMI Bridge
+	a011  N10 Family Integrated Graphics Controller
+	a012  N10 Family Integrated Graphics Controller
+	a013  N10 Family CHAPS counter
 	a620  6400/6402 Advanced Memory Buffer (AMB)
 	b152  21152 PCI-to-PCI Bridge
 		8086 b152  21152 PCI-to-PCI Bridge
