diff -u new//help.c old//help.c
--- new//help.c	2010-08-24 20:18:17.759957053 +0700
+++ old//help.c	2000-07-30 11:15:47.000000000 +0700
@@ -324,19 +324,11 @@
   while (current < num) { /* big while loop, to process all the charactes in buffer */
 
     /* look for the next escape character '~' */
-    while ((current < num) && (buf[current] != 28)) current++;
+    while ((current < num) && (buf[current] != 0x7e)) current++;
     /* and write the sequence befor esc char to the comm port */
-    if (current) {
-    	int qw = 0;
-    	for (qw = 0; qw < current; qw++)
-    	{
-    		if (buf[qw] == 10) buf[qw] = 13;
-    	}
-    	write (fd, buf, current);
-    }
+    if (current) write (fd, buf, current);
 
     if (current < num) { /* process an escape sequence */
-      
       /* found an escape character */
       if (help_state == 0)
 	help_escape();
diff -u new//microcom.c old//microcom.c
--- new//microcom.c	2010-08-24 20:33:33.675956788 +0700
+++ old//microcom.c	2000-08-27 22:22:47.000000000 +0700
@@ -24,7 +24,7 @@
 int crnl_mapping; //0 - no mapping, 1 mapping
 int script = 0; /* script active flag */
 char scr_name[MAX_SCRIPT_NAME] = "script.scr"; /* default name of the script */
-char device[MAX_DEVICE_NAME] = "/dev/"; /* serial device name */
+char device[MAX_DEVICE_NAME]; /* serial device name */
 int log = 0; /* log active flag */
 FILE* flog;   /* log file */
 int  pf = 0;  /* port file descriptor */
@@ -54,7 +54,10 @@
   /* set hardware flow control by default */
   pts->c_cflag |= CRTSCTS;
   pts->c_iflag &= ~(IXON | IXOFF | IXANY);
-//  pts->c_iflag |= ICRNL;
+  /* set 9600 bps speed by default */
+  cfsetospeed(pts, B9600);
+  cfsetispeed(pts, B9600);
+  
 }
 
 void init_stdin(struct termios *sts) {
@@ -95,8 +98,7 @@
 	  "           example: -D/dev/ttyS3\n"
 	  "    -S              run script from script.scr (default)\n"
 	  "    -Sscrfile       run script from scrfile\n\n"
-	  "microcom provides session logging in microcom.log file\n"
-	  "To enter in program menu press 'Ctrl'+'\\'\n");
+	  "microcom provides session logging in microcom.log file\n");
   fprintf(stderr, "Exitcode %d - %s %s\n\n", exitcode, str, dev);
   exit(exitcode);
 }
@@ -120,8 +122,7 @@
   struct sigaction sact;/* used to initialize the signal handler */
   int i;
   
-  if (argc == 1) main_usage(1, "no port name", "");
-  sprintf(device, "/dev/");
+  device[0] = '\0';
 
   /* parse command line */
   for (i = 1; i < argc; i++) {
@@ -137,19 +138,8 @@
       continue;
     }
     if (strncmp(argv[i], "-?", 2) == 0 ||
-    	strncmp(argv[i], "-h", 2) == 0 ||
-    	strncmp(argv[i], "--help", 2) == 0 ||
-    	strncmp(argv[i], "/?", 2) == 0 ||
-    	strncmp(argv[i], "?", 2) == 0 ||
-    	strncmp(argv[i], "help", 2) == 0 ||
 	strncmp(argv[i], "-H", 2) == 0)
-	{
       main_usage(0, "", "");
-      }
-      else
-      {
-       strncpy(&device[5], argv[i], MAX_DEVICE_NAME);
-      }
   }
    
 
@@ -171,7 +161,6 @@
     main_usage(2, "cannot open device", device);
 
 
-  fprintf(stderr, "To enter in program menu press \"Ctrl\"+\"\\\"\n");
   /* modify the port configuration */
   tcgetattr(pf, &pts);
   memcpy(&pots, &pts, sizeof(pots));
